### PR TITLE
python: added iio_attr with Python bindings.

### PR DESCRIPTION
**Pull request by cristi-iacob**
Implemented the iio_attr module with Python bindings.

Signed-off-by: Cristi Iacob <cristian.iacob@analog.com>

### CODE REVIEW

**Comment by rgetz** [comment]
some of the codacy warnings should go way when #460 is merged

**Comment by tfcollins** [comment]
Please add link in sphinx doc as well to this example like the others.

**Comment by cristi-iacob** [comment]
Done.

**Comment by dNechita** [comment]
Please resolve conflicts.

**Comment by cristi-iacob** [comment]
I have solved them. Although, @tfcollins suggested me to implement the python examples in a more pythonic fashion (and not on the same structure as the ones from C), so I think we can close this pull request and open another one with the new iio_attr.py.

**Comment by dNechita** [comment]
> I have solved them. Although, @tfcollins suggested me to implement the python examples in a more pythonic fashion (and not on the same structure as the ones from C), so I think we can close this pull request and open another one with the new iio_attr.py.

Why do you want to start a new PR and not continue to work on this one?

**Comment by cristi-iacob** [comment]
Ok, I will mark it as a draft until then. Thanks!

File: bindings/python/examples/iio_attr.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+        if channel.data_format.repeat > 1:
+            print("X" + str(channel.data_format.repeat), end="")
+
+        print(", index: " + str(channel.index) + ", format: "
+              + "b" if channel.data_format.is_be else "l"
+              + "e:" + sign + str(channel.data_format.bits)
+              + "/" + str(channel.data_format.length) + str(channel.data_format.repeat)
+              + ">>" + str(channel.data_format.shift))
+
+        print() if self.arguments.scan_only else print(", ", end="")
```
Comment (tfcollins) [inline bindings/python/examples/iio_attr.py:335]: To fix pep8 try this: `print("" if self.arguments.scan_only else ", ", end="")`

**Comment by tfcollins** [comment]
> Should I change the code so that all of them appear in init?

There are a couple things:
- If the definitions are not conditional I would put them in __init__ directly 
- Put None as the initial value in __init__ and set them with the current methods
- Or have the methods return tuples to set the relevant properties in __init__

File: bindings/python/examples/iio_attr.py (hunk 1 above)
Comment (cristi-iacob) [reply bindings/python/examples/iio_attr.py]: Solved.

**Comment by cristi-iacob** [comment]
I have solved the issues with the class attributes and with the print statements. Thanks for the suggestions!

**Comment by rgetz** [comment]
The last issues is an easy fix.
```
First line should end with a period (not 'n') (D400)
```

if you make the file:
```
#!/usr/bin/env python
"""
Copyright (C) 2015 Analog Devices, Inc.
```

like the others - it will be fine.

**Comment by cristi-iacob** [comment]
I have solved the last issue. Thanks for the suggestion!

**Comment by rgetz** [comment]
Thanks for all the updates. Looks good to me - will merge after @tfcollins has a quick look.

**Pull request merged**